File: Java/SASSample/app/src/main/java/com/smartadserver/android/sassample/BannerActivity.java
package com.smartadserver.android.sassample;

import android.os.Bundle;

import androidx.activity.EdgeToEdge;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import android.util.Log;
import android.view.View;
import android.webkit.WebView;
import android.widget.Button;

import com.smartadserver.android.library.model.SASAdElement;
import com.smartadserver.android.library.model.SASAdPlacement;
import com.smartadserver.android.library.ui.SASBannerView;
import com.smartadserver.android.library.ui.SASRotatingImageLoader;
import com.smartadserver.android.library.util.SASConfiguration;

/**
 * Simple activity featuring a banner ad.
 */

public class BannerActivity extends AppCompatActivity {

    /*****************************************
     * Ad Constants
     *****************************************/
    private final static int SITE_ID = 104808;
    private final static int PAGE_ID = 663262;
    private final static int FORMAT_ID = 15140;
    private final static String TARGET = "";

    // If you are an inventory reseller, you must provide your Supply Chain Object information.
    // More info here: https://help.smartadserver.com/s/article/Sellers-json-and-SupplyChain-Object
    private final static String SUPPLY_CHAIN_OBJECT_STRING = null; // "1.0,1!exchange1.com,1234,1,publisher,publisher.com";

    /*****************************************
     * Members declarations
     *****************************************/
    // Banner view (as declared in the main.xml layout file, in res/layout)
    SASBannerView mBannerView;

    // Handler class to be notified of ad loading outcome
    SASBannerView.BannerListener bannerListener;

    // Button declared in main.xml
    Button mRefreshBannerButton;


    /**
     * performs Activity initialization after creation
     */
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_banner);

        EdgeToEdge.enable(this);

        //Set Title
        setTitle(R.string.title_activity_banner);

        /*****************************************
         * now perform Ad related code here
         *****************************************/

        // Enable output to Android Logcat (optional)
        SASConfiguration.getSharedInstance().setLoggingEnabled(true);

        // Enable debugging in Webview if available (optional)
        WebView.setWebContentsDebuggingEnabled(true);

        // Initialize SASBannerView
        initBannerView();

        // Create button to manually refresh the ad
        mRefreshBannerButton = this.findViewById(R.id.reloadButton);
        mRefreshBannerButton.setOnClickListener(new View.OnClickListener() {
            public void onClick(View v) {
                loadBannerAd();
            }
        });

        // Load Banner ad
        loadBannerAd();

    }

    /**
     * Overriden to clean up SASAdView instances. This must be done to avoid IntentReceiver leak.
     */
    @Override
    protected void onDestroy() {
        mBannerView.onDestroy();
        super.onDestroy();
    }

    /**
     * Initialize the SASBannerView instance of this Activity
     */
    private void initBannerView() {
        // Fetch the SASBannerView inflated from the main.xml layout file
        mBannerView = this.findViewById(R.id.banner);

        // Add a loader view on the banner. This view covers the banner placement, to indicate progress, whenever the banner is loading an ad.
        // This is optional
        View loader = new SASRotatingImageLoader(this);
        loader.setBackgroundColor(getResources().getColor(R.color.colorLoaderBackground));
        mBannerView.setLoaderView(loader);

        bannerListener = new SASBannerView.BannerListener() {
            @Override
            public void onBannerAdLoaded(@NonNull SASBannerView bannerView, @NonNull SASAdElement adElement) {
                Log.i("Sample", "Banner loading completed");
            }

            @Override
            public void onBannerAdFailedToLoad(@NonNull SASBannerView bannerView, @NonNull Exception e) {
                Log.i("Sample", "Banner loading failed: " + e.getMessage());
            }

            @Override
            public void onBannerAdClicked(@NonNull SASBannerView bannerView) {
                Log.i("Sample", "Banner was clicked");
            }

            @Override
            public void onBannerAdExpanded(@NonNull SASBannerView bannerView) {
                Log.i("Sample", "Banner was expanded");
            }

            @Override
            public void onBannerAdCollapsed(@NonNull SASBannerView bannerView) {
                Log.i("Sample", "Banner was collapsed");
            }

            @Override
            public void onBannerAdResized(@NonNull SASBannerView bannerView) {
                Log.i("Sample", "Banner was resized");
            }

            @Override
            public void onBannerAdClosed(@NonNull SASBannerView bannerView) {
                Log.i("Sample", "Banner was closed");
            }

            @Override
            public void onBannerAdVideoEvent(@NonNull SASBannerView bannerView, int videoEvent) {
                Log.i("Sample", "Video event " + videoEvent + " was triggered on Banner");
            }
        };

        mBannerView.setBannerListener(bannerListener);
    }

    /**
     * Loads an ad on the banner
     */
    private void loadBannerAd() {
        SASAdPlacement adPlacement = new SASAdPlacement(SITE_ID,
                PAGE_ID,
                FORMAT_ID,
                TARGET,
                SUPPLY_CHAIN_OBJECT_STRING);

        // Load banner ad with appropriate parameters (siteID,pageID,formatID,master,targeting,adResponseHandler)
        mBannerView.loadAd(adPlacement);
    }

}

File: Kotlin/SASSample/app/src/main/java/com/smartadserver/android/kotlinsample/BannerActivity.kt
package com.smartadserver.android.kotlinsample

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat
import com.smartadserver.android.kotlinsample.databinding.ActivityBannerBinding
import com.smartadserver.android.library.model.SASAdElement
import com.smartadserver.android.library.model.SASAdPlacement
import com.smartadserver.android.library.ui.SASBannerView
import com.smartadserver.android.library.ui.SASRotatingImageLoader

/**
 * Simple activity featuring a banner ad.
 */
class BannerActivity : AppCompatActivity() {

    // If you are an inventory reseller, you must provide your Supply Chain Object information.
    // More info here: https://help.smartadserver.com/s/article/Sellers-json-and-SupplyChain-Object
    private val supplyChainObjectString: String? = null // "1.0,1!exchange1.com,1234,1,publisher,publisher.com";

    private val bannerPlacement = SASAdPlacement(104808, 663262, 15140, "", supplyChainObjectString)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // Binding object to retrieve UI elements
        val binding = ActivityBannerBinding.inflate(layoutInflater)
        setContentView(binding.root)

        // Add a loader view on the banner. This view covers the banner placement, to indicate profress, whenever the banner is loading an ad.
        // This is optional.
        binding.bannerView.loaderView = SASRotatingImageLoader(this).apply {
            setBackgroundColor(ContextCompat.getColor(context, R.color.colorLoaderBackground))
        }

        // Set the banner listener
        binding.bannerView.bannerListener = object : SASBannerView.BannerListener {
            override fun onBannerAdLoaded(banner: SASBannerView, adElement: SASAdElement) {
                Log.i("Sample", "Banner loading completed")
            }

            override fun onBannerAdFailedToLoad(banner: SASBannerView, e: Exception) {
                Log.i("Sample", "Banner loading failed: $e")
            }

            override fun onBannerAdClicked(banner: SASBannerView) {
                Log.i("Sample", "Banner was clicked")
            }

            override fun onBannerAdExpanded(banner: SASBannerView) {
                Log.i("Sample", "Banner was expanded")
            }

            override fun onBannerAdCollapsed(banner: SASBannerView) {
                Log.i("Sample", "Banner was collapsed")
            }

            override fun onBannerAdClosed(banner: SASBannerView) {
                Log.i("Sample", "Banner was closed")
            }

            override fun onBannerAdResized(banner: SASBannerView) {
                Log.i("Sample", "Banner was resized")
            }

            override fun onBannerAdVideoEvent(banner: SASBannerView, event: Int) {
                Log.i("Sample", "Video event $event was triggered on banner")
            }
        }


        // Set the button click listener
        binding.reloadButton.setOnClickListener { binding.bannerView.loadAd(bannerPlacement) }

        // Load the ad
        binding.bannerView.loadAd(bannerPlacement)
    }
}